File: desapp-groupB-backend/src/main/java/model/score/ScoreManager.java
package model.score;

import java.util.List;

import model.utils.Entity;

public class ScoreManager extends Entity {
	
	private static final long serialVersionUID = 532916824793782845L;

	private List<Score> scores;

	public ScoreManager(List<Score> someList) {
		super();
    	scores = someList;
    } 

    public Double minimumScoreAccepted() {
        return 4.0;
    }
    public Double maximumPossibleScore() {
        return 5.0;
    }

    public Double averageScore() {
        if(this.scores.isEmpty())
            return maximumPossibleScore();
        else
            return this.scores.stream().mapToDouble(Score::getValue).sum() / this.scores.size();
    }

    public void addScore(Score anyScore) {
        this.scores.add(anyScore);
    }
    
    public List<Score> getScores() {
		return scores;
	}

	public void setScores(List<Score> scores) {
		this.scores = scores;
	}
    
    public ScoreManager() {
		super();
    }
}

File: carpndWeb/src/app/models/score-manager.ts
import { Score } from './../models/score';

export class ScoreManager {
	scores: Score[];


averageScore():number{
	console.log( "sdahaskjdhaskjdhaskjd");
	console.log((this.scores.map((s) => s.value)));
	console.log( (this.scores.map((s) => s.value).reduce((a, b) => a+b)));
	return (this.scores.map((s) => s.value).reduce((a, b) => a+b))/ this.scores.length
}

}